src/cascadia/inc/cppwinrt_utils.h

```diff
@@ -55,4 +55,34 @@ private:
 // Use this if you have a Windows.Foundation.TypedEventHandler
 #define DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER(className, name, eventHandler, sender, args)                                                        \
     winrt::event_token className::name(Windows::Foundation::TypedEventHandler<sender, args> const& handler) { return eventHandler.add(handler); } \
-    void className::name(winrt::event_token const& token) noexcept { eventHandler.remove(token); }
\ No newline at end of file
+    void className::name(winrt::event_token const& token) noexcept { eventHandler.remove(token); }
+
+// This is a helper method for deserializing a SAFEARRAY of
+// COM objects and converting it to a vector that
+// owns the extracted COM objects
+        template<typename T>
+        std::vector<wil::com_ptr<T>> SafeArrayToOwningVector(SAFEARRAY* safeArray)
+        {
+            T** pVals;
+            THROW_IF_FAILED(SafeArrayAccessData(safeArray, (void**)&pVals));
+
+            long lBound, uBound;
+            THROW_IF_FAILED(SafeArrayGetLBound(safeArray, 1, &lBound));
+            THROW_IF_FAILED(SafeArrayGetUBound(safeArray, 1, &uBound));

```

Comment:
I know why you're not destructing here, but please leave comments: we can't destruct/release any of the elements in the safearray because we can't even figure out how many elements there ARE. There's nothing that we CAN safely destruct.